tools/write_release_and_log.py

```diff
@@ -0,0 +1,145 @@
+"""
+
+import os
+import sys
+import glob
+import subprocess
+from hashlib import md5
+from hashlib import sha256
+from pathlib import Path
+
+sys.path.insert(0, os.path.dirname(__file__))
+sys.path.insert(1, os.path.join(os.path.dirname(__file__), 'tools'))
+
+
+try:
+    version_utils = __import__("version_utils")
+    FULLVERSION = version_utils.VERSION
+    # This is duplicated from setup.py
+    if os.path.exists('.git'):
+        GIT_REVISION, _ = version_utils.git_version(
+            os.path.join(os.path.dirname(__file__), '..'))
+    else:
+        GIT_REVISION = "Unknown"
+
+    if not version_utils.ISRELEASED:
+        if GIT_REVISION == "Unknown":
+            FULLVERSION += '.dev0+Unknown'
+        else:
+            FULLVERSION += '.dev0+' + GIT_REVISION[:7]
+finally:
+    sys.path.pop(1)
+    sys.path.pop(0)
+
+try:
+    # Ensure sensible file permissions
+    os.umask(0o022)
+except AttributeError:
+    # No umask on non-posix
+    pass
+
+
+def get_latest_release_doc(path):
+    """
+    Method to Pick the latest file from 'doc/release'
+    """
+    files_path = os.path.join(path, '*')
+    files = sorted(
+        glob.iglob(files_path), key=os.path.getmtime, reverse=True)
+    return files[0]
+
+# ----------------------------
+# Release notes and Changelog
+# ----------------------------
+
+
+def compute_md5(idirs):
+    released = os.listdir(idirs)
+    checksums = []
+    for fn in sorted(released):
+        fn_updated = os.path.join("release", fn)
+        with open(fn_updated, 'rb') as f:
+            m = md5(f.read())
+        checksums.append('%s  %s' % (m.hexdigest(), os.path.basename(fn)))
+    return checksums
+
+
+def compute_sha256(idirs):
+    # better checksum so gpg signed README.txt containing the sums can be used
+    # to verify the binaries instead of signing all binaries
+    released = os.listdir(idirs)
+    checksums = []
+    for fn in sorted(released):
+        fn_updated = os.path.join("release", fn)
+        with open(fn_updated, 'rb') as f:
+            m = sha256(f.read())
+        checksums.append('%s  %s' % (m.hexdigest(), os.path.basename(fn)))
+
+    return checksums
+
+
+def write_release_task(filename='NOTES.txt'):
+    idirs = Path('release')
+    source = Path(RELEASE)
+    target = Path(filename)
+    if target.exists():
+        target.remove()
+
+    tmp_target = Path(filename + '.txt')
+    os.system(f'cp {source} {tmp_target}')
+
+    with open(str(tmp_target), 'a') as ftarget:
+        ftarget.writelines("""
+Checksums
+=========
+
+MD5
+~~~
+
+""")
+        ftarget.writelines(['%s\n' % c for c in compute_md5(idirs)])
+        ftarget.writelines("""
+SHA256
+~~~~~~
+
+""")
+        ftarget.writelines(['%s\n' % c for c in compute_sha256(idirs)])
+
+
+def write_log_task(filename='Changelog'):
+    st = subprocess.Popen(
+        ['git', 'log', '%s..%s' % (LOG_START, LOG_END)],
+        stdout=subprocess.PIPE)
+
+    out = st.communicate()[0].decode()
```

Comment:
You want to add verification of the return status here so that if the command errors, the script doesn't just blindly continue. Here's how you can check:
```
python tools/write_release_and_log.py v1.8.0 v1.8.1
fatal: ambiguous argument 'v1.8.0..v1.8.1': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Release Logs and README generated successfully
```
`v1.8.1` doesn't exist yet, so this should raise an exception and not write any files.

Reply:
I have added the verification based on `returncode`

Reply:
That looks good now, thanks